test/integration/CookieLangTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+
+namespace Wovnio\Wovnphp\Tests\Integration;
+
+require_once 'src/wovnio/wovnphp/CookieLang.php';
+require_once(__DIR__ . '/../helpers/TestUtils.php');
+
+use Wovnio\Test\Helpers\StoreAndHeadersFactory;
+use Wovnio\Test\Helpers\TestUtils;
+use Wovnio\Wovnphp\CookieLang;
+
+class CookieLangTest extends \PHPUnit_Framework_TestCase
+{
+    protected function setUp()
+    {
+        $this->sourceDir  = realpath(dirname(__FILE__) . '/../..');
+        $this->docRoot    = '/var/www/html';
+
+        TestUtils::cleanUpDirectory($this->docRoot);
+
+        // Copy WOVN.php
+        mkdir("{$this->docRoot}/WOVN.php");
+        exec("cp -rf {$this->sourceDir}/src {$this->docRoot}/WOVN.php/src");
+        copy("{$this->sourceDir}/htaccess_sample", "{$this->docRoot}/.htaccess");
+
+        // Set html-swapper mock
+        mkdir("{$this->docRoot}/v0");
+        copy("{$this->sourceDir}/test/fixtures/integration/v0/translation", "{$this->docRoot}/v0/translation");
+    }
+
+    protected function tearDown()
+    {
+        TestUtils::cleanUpDirectory($this->docRoot);
+    }
+
+    public function testShould302RedirectDefaultLangQueryPattern()
```

Comment:
I generally don't like test names that don't describe the actual scenario and force you to read the test body. Unfortunately this is the current pattern of the existing tests in wovn.php

This is testing that when browsing to a default lang url with a wovn cookie, a redirection is sent.
`testRequestToDefaultLang__WithCookie__ShouldRedirect` something like this would be my choice here.

Similarly the ShouldNotRedirect tests are more like
`testRequestToTargetLang__WithCookie__ShouldNotRedirect`

I'm having a hard time figuring out what `testShouldNotRedirectDefaultLangToOtherLangPathPattern` tests by the name.
It's testing that a request to a target language with default lang cookie doesn't redirect to the cookie lang? Not obvious at all from the name.

Reply:
New names adopts a requested lang + cookie status + url pattern + expected behavior format. Should be friendlier now.